packages/marshal/src/helpers/copyRecord.js

```diff
@@ -24,25 +24,24 @@ export const CopyRecordHelper = harden({
   styleName: 'copyRecord',
 
   canBeValid: (candidate, check) => {
+    const reject = details => check(false, details);
     const proto = getPrototypeOf(candidate);
     if (proto !== objectPrototype && proto !== null) {
-      return check(false, X`Unexpected prototype for: ${candidate}`);
+      return reject(X`Unexpected prototype for: ${candidate}`);
     }
     const descs = getOwnPropertyDescriptors(candidate);
     const descKeys = ownKeys(descs);
 
     for (const descKey of descKeys) {
       if (typeof descKey !== 'string') {
         // Pass by copy
-        return check(
-          false,
+        return reject(
           X`Records can only have string-named own properties: ${candidate}`,
         );
```

Comment:
@erights I’m reasonably certain that the X/describe tagged literal does not have the same effect on `reject` or even as an argument to the `Error` constructor as it does on `assert` methods. You may need `assert.error` here for the details to be extractable by the console.

Reply:
`assertValid` calls `canBeValid(candidate, assertChecker)`, where `assertChecker` (and therefore `check` here) is `(cond, details) => { assert(cond, details); return true; }`, so this `reject` ultimately does send output from ``` X`…` ``` to `assert` in that case. Or are you saying that there's some dependence upon e.g. static analysis detecting literal use of "assert"?

Reply:
No, my review was too shallow. I see now you’re plumbing through `assert` and that `reject` is not `Promise.reject`.